### DISCUSSION TITLE
TTFB on Server Side Rendering

### DISCUSSION THREAD

**Discussion by doptster** [vercel/next.js#23133] [Help]
I am running NextJS in Docker.
When running the project in localhost, the ttfb is only around 40ms difference with the **getInitialProps** time, for example it took 300ms to complete **getInitialProps**, and ttfb is around 340ms.

However, when running the project under docker the gaps between server fetch time and ttfb is increased to 600ms, for example it took 300ms to complete  **getInitialProps**, and ttfb is around 900ms. 

When we comment out the major components rendering the gaps lowered to decent amount (~100ms), does anyone has idea on explaining the additional gaps between server fetch time and ttfb? What could be the possible root cause for it

**Comment by akellbl4** [comment]
Do you have limitations for docker? Maybe the Next.js instance inside a docker container has fewer resources and handles requests slower.

**Reply by doptster** [reply]
I have also suspecting that, currently the docker is also running with nginx, I am working on to check those settings.
It just makes not much sense when the server time has only 300ms, but ttfb is 800ms++. Just wanna make sure there's nothing else could impact the ttfb from NextJS's perspective

**Reply by akellbl4** [reply]
I think if you don't have such delay without docker all overhead somewhere around docker.
Also, it's pretty obvious when you have many components for rendering your TTFB increases because SSR gets more work to do :)

**Reply by doptster** [reply]
Besides docker, since it's working good in localhost, do you think rendering too many data & components could still possibly be the issue? let's say I am rendering 200 of thumbnails based on the API results fetched in server side.

**Reply by akellbl4** [reply]
More components = more rendering time :) Also, It depends on how complex your components.
If I have such an amount for a render I prefer to render the first 10 on the server and lazy load (infinity scroll) for the rest.

**Reply by doptster** [reply]
Let's us check further if reducing server side components could work to reduce TTFB, thanks for the input.
